README.md

```diff
@@ -1,160 +1,104 @@
-```
-
-Handles JSON as binary or string. Uses the supplied functions
-JsonDecode to parse the JSON request and JsonEncode to encode the reply as JSON.
+```erlang
+1> Body = ~'{"jsonrpc": "2.0", "method": "foo", "params": [1,2,3], "id": 1}'.
+<<"{\"jsonrpc\": \"2.0\", \"method\": \"foo\", \"params\": [1,2,3], \"id\": 1}">>
+
+2> Handler = fun(~"foo", Params) -> {ok, lists:reverse(Params)}; (_, _) -> {error, method_not_found} end.
+#Fun<erl_eval.41.81571850>
+
+3> jsonrpc2:handle(Body, Handler).
+{reply,["{",
+        [[34,<<"id">>,34],58|<<"1">>],
+        [[44,
+          [34,<<"result">>,34],
+          58,91,<<"3">>,44,<<"2">>,44,<<"1">>,93],
+         [44,[34,<<"jsonrpc">>,34],58,34,<<"2.0">>,34]],
+        "}"]}
+
+4> jsonrpc2:handle(~"dummy", Handler).
+=WARNING REPORT==== 10-Aug-2025::13:58:12.336132 ===
+Failed to decode request in JSON format: error {invalid_byte,100}
+{reply,["{",
+        [[34,<<"error">>,34],
+         58,"{",
+         [[34,<<"code">>,34],58|<<"-32700">>],
+         [[44,[34,<<"message">>,34],58,34,<<"Parse error.">>,34]],
+         "}"],
+        [[44,[34,<<"id">>,34],58|<<"null">>],
+         [44,[34,<<"jsonrpc">>,34],58,34,<<"2.0">>,34]],
+        "}"]}
+
+5> {reply, R} = jsonrpc2:handle(~'{"x": 42}', Handler).
+{reply,["{",
+        [[34,<<"error">>,34],
+         58,"{",
+         [[34,<<"code">>,34],58|<<"-32600">>],
+         [[44,[34,<<"message">>,34],58,34,<<"Invalid Request.">>,34]],
+         "}"],
+        [[44,[34,<<"id">>,34],58|<<"null">>],
+         [44,[34,<<"jsonrpc">>,34],58,34,<<"2.0">>,34]],
+        "}"]}
+
+6> iolist_to_binary(R).
+<<"{\"error\":{\"code\":-32600,\"message\":\"Invalid Request.\"},\"id\":null,\"jsonrpc\":\"2.0\"}">>
+
+
+7> Term = #{~"jsonrpc" => ~"2.0", ~"method" => ~"foo", ~"params" => [1,2,3], ~"id" => 1}.
+#{<<"id">> => 1,<<"jsonrpc">> => <<"2.0">>,
+  <<"method">> => <<"foo">>,
+  <<"params">> => [1,2,3]}
+
+8> jsonrpc2:handle_term(Term, Handler).
+{reply,#{id => 1,result => [3,2,1],jsonrpc => <<"2.0">>}}
 
-```Erlang
-handle(Req::term(), handlerfun(), mapfun(), JsonDecode::fun(),
-    JsonEncode::fun()) -> {reply, term()} | noreply
 ```
 
-Like `handle/4`, but also takes a map function for batch
-processing. See `handle/3` above.
 
-Error Handling
+Client example
 --------------
 
-A requests that is not valid JSON results in a "Parse error" JSON-RPC response.
-
-An invalid JSON-RPC request (though valid JSON) results in an "Invalid Request"
-response. In these two cases the handler callback function is never called.
-
-To produce an error response from the handler function, you may throw one of
-the exceptions below. They will be caught and turned into a corresponding
-JSON-RPC error response.
-
-  * `throw(method_not_found)` is reported as "Method not found" (-32601)
-  * `throw(invalid_params)` is reported as "Invalid params" (-32602)
-  * `throw(internal_error)` is reported as "Internal error" (-32603)
-  * `throw(server_error)` is reported as "Server error" (-32000)
```

Comment:
Returning RPC errors from the handler using throw is not supported anymore? Why not?